### PR TITLE
feat(license): add acknowledgements to license clearing

### PR DESCRIPTION
**Pull request by shaheemazmalmmd**
- upload a package and go to single file clearing.
- you should find new acknowledgement TD.
- add new acknowledgement to the file and make it identified.
- Generate READMEOSS and Unified Report.
- check if the acknowledgements exists

### CODE REVIEW

File: src/lib/php/Data/Clearing/ClearingEvent.php [hunk 1]
```diff
@@ -46,6 +46,10 @@ class ClearingEvent extends Object implements LicenseClearing
    * @param int $groupId
    * @param int $eventType
    * @param ClearingLicense $clearingLicense
+   * @param boolean $removed
+   * @param string $reportinfo
+   * @param string $acknowledgement
+   * @param string $comment
```
Comment (ag4ums) [inline src/lib/php/Data/Clearing/ClearingEvent.php]: $removed, $reportinfo, $acknowledgement, $comment are these require

File: src/lib/php/Dao/ClearingDao.php [hunk 2]
```diff
@@ -852,17 +866,28 @@ public function getMainLicenseIds($uploadId, $groupId)
...
     $this->dbManager->insertTableRow('upload_clearing_license',
             array('upload_fk'=>$uploadId,'group_fk'=>$groupId,'rf_fk'=>$licenseId));
   }
-  
+
+  /**
+   * @param uploadId
+   * @param int $groupId
+   * @param int $licenseId
+   */
```
Comment (ag4ums) [inline src/lib/php/Dao/ClearingDao.php]: please also add return param

File: src/readmeoss/agent/readmeoss.php [hunk 3]
```diff
@@ -120,27 +131,61 @@ private function writeReport($contents, $uploadId)
...
+   * @param $extract
+   * @param $break
+   */ 
+  private function createReadMeOSSFormat($addSeparator, $dataForReadME, $extract='text', $break)
+  {
+    $outData = "";
+    foreach($dataForReadME as $statements){
+     $outData .= $statements[$extract] . $break;
+     if(!empty($addSeparator)){
+       $outData .= $addSeparator . $break;
```
Comment (ag4ums) [inline src/readmeoss/agent/readmeoss.php]: please correct the alignment

**Comment by mcjaeger** [comment]
@ag4ums would you mind to clarify if the review comments are covered.

Also, I suspect that the same error shown as in https://github.com/fossology/fossology/pull/980 has also the same cause.

**Comment by ag4ums** [comment]
now this is fixed with the pull request #1063

**Comment by NicolasToussaint** [comment]
Do you think that the Acknowledgement information could also be exported to the SPDX reports ?

It seems to me that the ["File Contributor"](https://spdx.org/spdx_specification_2_0_html#h.sqyw64) entry in the SPDX 2.0 Specification would be a good candidate, what do you think ?

**Comment by mcjaeger** [comment]
I think the issue is here at the spdx spec: https://github.com/spdx/spdx-spec/issues/28

I am not sure if FileContributor is the intended option here. Maybe as an interim solution it could be done. However, I am not sure we should build FOSSology (and SW360) to have it in FileContributor then if there is a new field planned. @kestewart would you have an update on this?

**Comment by NicolasToussaint** [comment]
Good spot, though I'm a bit confused on how these fields should be used.
About the AttributionText field you refer to:
> It can be considered as a combination of information found in properties like FileCopyrightText, LicenseInfoInFile, FileContributor, but it might not simply the sum of these values.

Anyhow, I guess I would be fine with a temporary export to the FileContributor.

**Comment by mcjaeger** [comment]
Hm, I see the point that this field is "partially matching" maybe it could be good to have the acknowledgement there for the first. It think it depends, how likely the spec is updated within the next three months, for example ...

**Pull request merged**